### PR TITLE
Fix refresh for skipChatButtonRendering

### PR DESCRIPTION
**Pull request by marghayk**
- When skipChatButtonRendering prop is set to true and we refresh the chat, it starts a new chat instead of connecting to the same chat. This fixes this issue.
- Upgrading components

### CODE REVIEW

File: chat-widget/src/components/livechatwidget/livechatwidgetstateful/LiveChatWidgetStateful.tsx [hunk 1]
```diff
@@ -187,7 +187,10 @@ export const LiveChatWidgetStateful = (props: ILiveChatWidgetProps) => {
                             eventName: BroadcastEvent.StartChatSkippingChatButtonRendering,
                         };
                         BroadcastService.postMessage(chatStartedSkippingChatButtonRendering);
-                        setPreChatAndInitiateChat(chatSDK, props.chatConfig, props.getAuthToken, dispatch, setAdapter);
+                        const optionalParams = (!isUndefinedOrEmpty(state.domainStates?.liveChatContext) && state.appStates.conversationState === ConversationState.Active) ?
+                            { liveChatContext: state.domainStates?.liveChatContext } :
```
Comment (dishakhattri) [inline chat-widget/src/components/livechatwidget/livechatwidgetstateful/LiveChatWidgetStateful.tsx:191]: Wouldn't it get reset at line https://github.com/microsoft/omnichannel-chat-widget/blob/c9a643a4fb5d8c219f3b408b52b7e4f5d8919199/chat-widget/src/components/livechatwidget/common/startChat.ts#L75 ?

File: chat-widget/src/components/livechatwidget/common/startChat.ts [hunk 2]
```diff
@@ -59,7 +59,7 @@ const prepareStartChat = async (props: ILiveChatWidgetProps, chatSDK: any, state
 };
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
-const setPreChatAndInitiateChat = async (chatSDK: any, chatConfig: ChatConfig | undefined, getAuthToken: ((authClientFunction?: string) => Promise<string | null>) | undefined, dispatch: Dispatch<ILiveChatWidgetAction>, setAdapter: any, isProactiveChat?: boolean | false, proactiveChatEnablePrechatState?: boolean | false) => {
+const setPreChatAndInitiateChat = async (chatSDK: any, chatConfig: ChatConfig | undefined, getAuthToken: ((authClientFunction?: string) => Promise<string | null>) | undefined, dispatch: Dispatch<ILiveChatWidgetAction>, setAdapter: any, isProactiveChat?: boolean | false, proactiveChatEnablePrechatState?: boolean | false, params?: any) => {
     // Getting prechat Survey Context
     const parseToJson = false;
     const preChatSurveyResponse: string = await chatSDK.getPreChatSurvey(parseToJson);
```
Comment (abhinab22mohanty) [inline chat-widget/src/components/livechatwidget/common/startChat.ts:65]: Also wouldn't the preChat be loaded again?

https://github.com/microsoft/omnichannel-chat-widget/blob/c9a643a4fb5d8c219f3b408b52b7e4f5d8919199/chat-widget/src/components/livechatwidget/common/startChat.ts#L70

File: chat-widget/src/components/livechatwidget/livechatwidgetstateful/LiveChatWidgetStateful.tsx (hunk 1 above)
Comment (marghayk) [reply chat-widget/src/components/livechatwidget/livechatwidgetstateful/LiveChatWidgetStateful.tsx]: Fixed

File: chat-widget/src/components/livechatwidget/common/startChat.ts (hunk 2 above)
Comment (marghayk) [reply chat-widget/src/components/livechatwidget/common/startChat.ts]: Fixed

File: chat-widget/src/components/livechatwidget/livechatwidgetstateful/LiveChatWidgetStateful.tsx [hunk 3]
```diff
@@ -187,7 +187,13 @@ export const LiveChatWidgetStateful = (props: ILiveChatWidgetProps) => {
...
                         };
                         BroadcastService.postMessage(chatStartedSkippingChatButtonRendering);
-                        setPreChatAndInitiateChat(chatSDK, props.chatConfig, props.getAuthToken, dispatch, setAdapter);
+                        if (!isUndefinedOrEmpty(state.domainStates?.liveChatContext)
+                            && state.appStates.conversationState === ConversationState.Active) {
+                            const optionalParams = { liveChatContext: state.domainStates?.liveChatContext };
+                            initStartChat(chatSDK, props.chatConfig, props.getAuthToken, dispatch, setAdapter, optionalParams);
+                        } else {
+                            setPreChatAndInitiateChat(chatSDK, props.chatConfig, props.getAuthToken, dispatch, setAdapter, undefined, undefined);
+                        }
```
Comment (sarojkpr) [inline chat-widget/src/components/livechatwidget/livechatwidgetstateful/LiveChatWidgetStateful.tsx:196]: Do we really need to pass undefined here, those are optional params?

**Review by sarojkpr — APPROVED** [review]

File: chat-widget/src/components/livechatwidget/livechatwidgetstateful/LiveChatWidgetStateful.tsx (hunk 3 above)
Comment (marghayk) [reply chat-widget/src/components/livechatwidget/livechatwidgetstateful/LiveChatWidgetStateful.tsx:196]: Removed

Comment (marghayk) [reply chat-widget/src/components/livechatwidget/livechatwidgetstateful/LiveChatWidgetStateful.tsx:196]: Removed

**Pull request merged**